### ISSUE TITLE
Properly escape list-like strings in `esc_format`

### ISSUE THREAD

**Issue by exhumea** [awesmubarak/markdown_strings#11]
I would have expected the following behavior for `esc_format`:

|input|expected|current|
|-|-|-|
|`'- One\n- Two\n'`|`'\- One\n\- Two\n'`|not escaped|
|`'* One\n* Two\n'`|`'\* One\n\* Two\n'`|OK|
|`'2024. This year...'`|`'2024\. This year...'`|not escaped|
|`> Not\n> Citing\n'`|`'\> Not\n\> Citing\n'`|not escaped|

**Comment by awesmubarak** [comment]
Yes you're right. I'm also realising that there might be a few other exceptions that `esc_format` doesn't cover (already escaped characters, code, maths,  URLs and email addresses). I'll do some research, some thinking, and some coding and push an update soon. Thanks for raising the issue.
